artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java

```diff
@@ -556,6 +556,12 @@ public LargeServerMessage createLargeMessage(final long id, final Message messag
             final int messageEncodeSize = largeMessage.getEncodeSize();
 
             if (messageEncodeSize > maxRecordSize) {
+               if (logger.isDebugEnabled()) {
+                  logger.debug("Message header too large for " + largeMessage);
+               } else {
+                  logger.warn("Message with id " + largeMessage.getMessageID() + " has a header too large. Set log level to debug to log the message header.");
```

Comment:
two things:

- the logger.warn should always be called, no matter what.
- Can you use a Logger with an ID here?